### ISSUE TITLE
Remote session on local VM unable to read shared folder on host

### ISSUE THREAD

**Issue by Hong-Revo** [Microsoft/RTVS#3487]
So, I've (finally!) managed to get file sharing working between a Hyper-V VM and the host. 

In Explorer on the VM, I can navigate to a shared folder on the host, and see that the files are there. I have to use an IP address because I'm a network admin noob.

![image](https://cloud.githubusercontent.com/assets/11328890/25653424/255d1e52-3030-11e7-91b8-ed421f9b3a1e.png)

If I run the following in rgui.exe inside the VM, I get a similar listing of the files:
```
> dir("\\\\169.254.119.118\\misc")
 [1] "cases.R"                                    
 [2] "clmSamp.xdf"                                
 [3] "dsvm"                                       
 [....]
```

However, if I run the same command in a remote RTVS session, I get nothing:
```
⚹> dir("\\\\169.254.119.118\\misc")
character(0)
```
Both the VM and host are running Win 10 Pro build 1703.

**Comment by int19h** [comment]
Since the code behind `dir` is the same in both cases (we don't do anything to handle it specially), it is likely that the difference is due to configuration. Specifically, I'm thinking it might be a firewall rule issue. While both things should use the same R.dll binary, firewall rules are applied to applications (i.e. .exe files). So if RGui.exe has a firewall rule for it, but Microsoft.R.Host.exe does not, it might be that the latter is simply not allowed to make the remote connection necessary to access files?

**Comment by Hong-Revo** [comment]
I couldn't find a firewall rule on the VM for rgui.exe. I turned it off anyway just to make sure, but the problem still occurs.

**Comment by karthiknadig** [comment]
Were you logged-in on the VM when you tried `dir` with RTVS Remote? If so, can you log out of the VM and (reset the R session) trying the same? Let me know the results. This will help narrow the issue.

**Comment by Hong-Revo** [comment]
Yes, I was logged in. I logged out and tried it again, after restarting VS. Still happens.

**Comment by Hong-Revo** [comment]
Another data point: if I use a nonsense location like `dir("\\\\123.123.123.123\\foo")`, there's a long delay before R returns. Whereas with this, it's immediate.

**Comment by Hong-Revo** [comment]
I had to mess around with the network config in the VM before I could get everything to work. I've no idea if this is the right way to do it.

- There's an external switch hooked up to my Wifi, for internet access
- An internal switch is what I use for host file sharing. I use the IP address because the computer name is taken by the external switch.

IP configuration in the VM:
```
C:\Users\hong>ipconfig

Windows IP Configuration


Ethernet adapter Ethernet 2:

   Connection-specific DNS Suffix  . :
   Link-local IPv6 Address . . . . . : fe80::a8da:eb37:28b7:dff1%9
   Autoconfiguration IPv4 Address. . : 169.254.223.241
   Subnet Mask . . . . . . . . . . . : 255.255.0.0
   Default Gateway . . . . . . . . . :

Ethernet adapter Ethernet:

   Connection-specific DNS Suffix  . :
   Link-local IPv6 Address . . . . . : fe80::2407:f617:8863:87d8%5
   IPv4 Address. . . . . . . . . . . : 192.168.0.2
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Default Gateway . . . . . . . . . : 192.168.0.1
```

And on the host:
```
C:\Users\hongooi>ipconfig

Windows IP Configuration


Ethernet adapter vEthernet (Internal Switch):

   Connection-specific DNS Suffix  . :
   Link-local IPv6 Address . . . . . : fe80::4cd0:f133:581b:7776%19
   Autoconfiguration IPv4 Address. . : 169.254.119.118
   Subnet Mask . . . . . . . . . . . : 255.255.0.0
   Default Gateway . . . . . . . . . :

Ethernet adapter vEthernet (Wifi Ex):

   Connection-specific DNS Suffix  . :
   Link-local IPv6 Address . . . . . : fe80::3434:a5a0:7f79:bfd8%21
   IPv4 Address. . . . . . . . . . . : 192.168.0.7
   Subnet Mask . . . . . . . . . . . : 255.255.255.0
   Default Gateway . . . . . . . . . : 192.168.0.1
```
